src/registry/routes/component-info.js

```diff
@@ -4,61 +4,77 @@ var parseAuthor = require('parse-author');
 var _ = require('underscore');
 
 var urlBuilder = require('../domain/url-builder');
+var getComponentInfoOrPreviewFallback = require('./helpers/get-component-fallback').getComponentInfoOrPreviewFallback;
 
-module.exports = function(repository){
-  return function(req, res){
+function getParams(component) {
+  var params = {};
+  if(!!component.oc.parameters){
+    var mandatoryParams = _.filter(_.keys(component.oc.parameters), function(paramName){
+      var param = component.oc.parameters[paramName];
+      return !!param.mandatory && !!param.example;
+    });
+
+    params = _.mapObject(_.pick(component.oc.parameters, mandatoryParams), function(param){
+      return param.example;
+    });
+  }
 
-    repository.getComponent(req.params.componentName, req.params.componentVersion, function(err, component){
+  return params;
+}
 
-      if(err){
-        res.errorDetails = err;
-        return res.status(404).json({ err: err });
+function getParsedAuthor(component) {
+  var author = component.author || {};
+  return _.isString(author) ? parseAuthor(author) : author;
+}
+
+function addGetRepositoryUrlFunction(component) {
+  component.getRepositoryUrl = function() {
+    if (_.isObject(this.repository)) {
+      if (this.repository.url) {
+        return this.repository.url;
       }
+    }
+    if (_.isString(this.repository)) {
+      return this.repository;
+    }
+    return null;
+  };
+}
 
-      var isHtmlRequest = !!req.headers.accept && req.headers.accept.indexOf('text/html') >= 0;
+module.exports = function(conf, repository){
+  return function(req, res){
 
-      if(isHtmlRequest && !!res.conf.discovery){
+    repository.getComponent(req.params.componentName, req.params.componentVersion, function(localRegistryError, localComponent){
 
-        var params = {},
-            author = component.author || {},
-            parsedAuthor = _.isString(author) ? parseAuthor(author) : author;
+      getComponentInfoOrPreviewFallback(conf, req, res, localRegistryError, localComponent, function(err, component) {
```

Comment:
this is the function that will try to do a fallback call to registry url

I've also looked briefly at replacing `repository.getComponent` with `getcomponenthelper`, however, helper returned very different format of response (that I would not be able to map).